src/Controller/EnMarche/CommitteeController.php

```diff
@@ -144,7 +144,9 @@ public function timelineAction(Request $request, Committee $committee): Response
         return $this->render('committee/timeline/feed.html.twig', [
             'committee' => $committee,
             'committee_timeline' => $timeline,
+            'committee_timeline_forms' => $this->createTimelineDeleteForms($timeline),
             'has_role_adherent' => $this->getUser() instanceof Adherent && $this->getUser()->isAdherent(),
+            'has_role_user' => $this->isGranted('ROLE_USER'),
```

Comment:
This should not live in the controller; Twig environment has registered functions for this. (like `is_granted(xxx)`)

I see `has_role_adherent` was already badly used here.

Please remove `has_role_adherent` and `has_role_user` and retrieve these informations directly in the template.
See: https://github.com/EnMarche/en-marche.fr/blob/develop/templates/base.html.twig#L9

Reply:
Not possible

Reply:
This template is used with ajax call and include call :s

Reply:
indeed, my bad.
this template is used as an inclusion in the layout and also as a single template.

Reply:
do we really need to call `createTimelineDeleteForms` and generate N forms for delete each feed item if the current user is not a adherent (hasn't ROLE_USER)?

what about check the role in the controller and build the forms if user has the good role, and in the template check only if `committee_timeline_forms` is defined?

Reply:
To create delete form, I can have a twig extension like create_delete_form(form) and in twig template have
```twig
{% set delete_form = create_delete_form(path('app_committee_timeline_delete', { id: item.id, slug: item.committee.slug }), 'committee_feed_delete') %}
{{ form_start(committee_timeline_forms[item.id]) }}
{{ form_widget(committee_timeline_forms[item.id].delete, { attr: { class: 'link--deletered'}}) }}
{{ form_end(committee_timeline_forms[item.id]) }}
```